internal/stickysessions/main.go

```diff
@@ -78,19 +83,20 @@ func NewStickySessionBalancer(service string, namespace string, containerPortNam
 		}
 	}
 
-	slices, err := endpointInformer.Lister().List(nil)
+	slices, err := cache.List(labels.Everything())
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	log.Printf("Endpoint slices found: %v", slices)
+
 	endpointStore := NewEndpointStoreFromEndpointSlices(slices, containerPortName)
 
-	cache := endpointInformer.Lister()
-	watcher := endpointInformer.Informer()
 	balancer := StickySessionBalancer{
 		Service:           service,
 		Namespace:         namespace,
 		ContainerPortName: containerPortName,
-		CookieName:        "reverse-proxy-sticky-session",
+		CookieName:        fmt.Sprintf("reverse-proxy-sticky-session-%s", service),
```

Comment:
I know this is still a WIP. But when I was testing stuff out today I saw something that will cause trouble.

Namely we need to set here a `CookiePath` parameter for the cookie. Because without it the cookies created have a Path based on the request that was sent. So you end up getting a different sticky session cookie per path.

This is a shortcoming that was present in the code from before. I just did not see it because I did not have a chance to test it out fully. This is where the `CookiePath` parameter from here should be set in the cookie https://github.com/SwissDataScienceCenter/renku-gateway/blob/master/internal/stickysessions/main.go#L128